extensions/wikia/PortableInfobox/services/PortableInfoboxRenderService.class.php

```diff
@@ -48,49 +20,33 @@ public static function getTemplates() {
 	 *
 	 * @param $theme
 	 * @param $layout
+	 * @param $accentColor
+	 * @param $accentColorText
 	 * @return string - infobox HTML
 	 */
-	public function renderInfobox( array $infoboxdata, $theme, $layout ) {
+	public function renderInfobox( array $infoboxdata, $theme, $layout, $accentColor, $accentColorText ) {
 		wfProfileIn( __METHOD__ );
 
 		$helper = new PortableInfoboxRenderServiceHelper();
+		$this->inlineStyles = $this->getInlineStyles( $accentColor, $accentColorText );
 		$infoboxHtmlContent = '';
-		$heroData = [ ];
 
-		// decide on image width
-		$this->imagesWidth = $helper->isMobile() ?
-			PortableInfoboxRenderServiceHelper::MOBILE_THUMBNAIL_WIDTH :
-			// if europa go with bigger images! else default size
-			$helper->isEuropaTheme() ? PortableInfoboxRenderServiceHelper::EUROPA_THUMBNAIL_WIDTH :
-				PortableInfoboxRenderServiceHelper::DEFAULT_DESKTOP_THUMBNAIL_WIDTH;
+		// decide on image width, if europa go with bigger images! else default size
+		$this->imagesWidth = $helper->isEuropaTheme() ? PortableInfoboxRenderServiceHelper::EUROPA_THUMBNAIL_WIDTH :
+			PortableInfoboxRenderServiceHelper::DEFAULT_DESKTOP_THUMBNAIL_WIDTH;
 
 		foreach ( $infoboxdata as $item ) {
-			$data = $item[ 'data' ];
 			$type = $item[ 'type' ];
 
-			switch ( $type ) {
-				case 'group':
-					$infoboxHtmlContent .= $this->renderGroup( $data );
-					break;
-				case 'navigation':
-					$infoboxHtmlContent .= $this->renderItem( 'navigation', $data );
-					break;
-				default:
-					if ( $helper->isMobile() && $helper->isValidHeroDataItem( $item, $heroData ) ) {
-						$heroData[ $type ] = $data;
-						continue;
-					}
+			if ( $this->templateEngine->isSupportedType( $type ) ) {
+				if ( $type === 'title' || $type === 'header' ) {
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Define a constant instead of duplicating this literal "header" 3 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1192)